source/overloading.tex

```diff
@@ -3489,24 +3470,23 @@
 \indextext{overloading!unary operator}
 
 \pnum
-A prefix unary operator shall be implemented by a
-non-static member function\iref{class.mfct} with no parameters or a
-non-member function with one parameter.
+For a \grammarterm{unary-operator} \tcode{@},
+an operator function \tcode{\keyword{operator}@} shall be
+a non-static member function\iref{class.mfct} with no parameters or
+a non-member function with one parameter.
```

Comment:
I think this rephrasing in terms of *unary-operator* is problematic: it suggests that (eg) `operator+` shall not have more than one parameter (including the implicit object parameter), which is incorrect.

The prior phrasing ("a prefix unary operator" here, "a binary operator" below) seems to avoid this problem if we add another rule that "an operator function shall declare a prefix unary operator, a binary operator, ...[other cases]...").

Reply:
Fixed.

I just noticed that [over.ass] syntactically appears to talk about copy-assignment only. It should probably talk about @= instead.

Reply:
I think it's fine and probably appropriate that we treat `@=` as binary operators and `=` as its own thing, because `operator@=` can be a member or non-member whereas `operator=` can only be a member. Though maybe "Simple assignment" would be a better subclause title than "Assignment"?

Reply:
Yes, definitely. Fixed.